src/contracts/core/AVSDirectory.sol

```diff
@@ -52,73 +60,275 @@ contract AVSDirectory is
+            );
+
+            // Migrate operator to operator sets.
+            _registerToOperatorSets(operators[i], msg.sender, operatorSetIds[i]);
+
+            // Deregister operator from AVS - this prevents the operator from being migrated again since
+            // the AVS can no longer use the legacy M2 registration path
+            avsOperatorStatus[msg.sender][operators[i]] = OperatorAVSRegistrationStatus.UNREGISTERED;
+            emit OperatorAVSRegistrationStatusUpdated(
+                operators[i], msg.sender, OperatorAVSRegistrationStatus.UNREGISTERED
+            );
+            emit OperatorMigratedToOperatorSets(operators[i], msg.sender, operatorSetIds[i]);
+        }
+    }
+
+    /**
+     *  @notice Called by AVSs to add an operator to a list of operatorSets.
+     *
+     *  @param operator The address of the operator to be added to the operator set.
+     *  @param operatorSetIds The IDs of the operator sets.
+     *  @param operatorSignature The signature of the operator on their intent to register.
+     *
+     *  @dev msg.sender is used as the AVS.
+     *  @dev The operator must not have a pending deregistration from the operator set.
+     */
+    function registerOperatorToOperatorSets(
         address operator,
+        uint32[] calldata operatorSetIds,
         ISignatureUtils.SignatureWithSaltAndExpiry memory operatorSignature
-    ) external onlyWhenNotPaused(PAUSED_OPERATOR_REGISTER_DEREGISTER_TO_AVS) {
-        require(
-            operatorSignature.expiry >= block.timestamp,
-            "AVSDirectory.registerOperatorToAVS: operator signature expired"
-        );
-        require(
-            avsOperatorStatus[msg.sender][operator] != OperatorAVSRegistrationStatus.REGISTERED,
-            "AVSDirectory.registerOperatorToAVS: operator already registered"
-        );
-        require(
-            !operatorSaltIsSpent[operator][operatorSignature.salt],
-            "AVSDirectory.registerOperatorToAVS: salt already spent"
-        );
-        require(
-            delegation.isOperator(operator),
-            "AVSDirectory.registerOperatorToAVS: operator not registered to EigenLayer yet"
-        );
-
-        // Calculate the digest hash
-        bytes32 operatorRegistrationDigestHash = calculateOperatorAVSRegistrationDigestHash({
-            operator: operator,
-            avs: msg.sender,
-            salt: operatorSignature.salt,
-            expiry: operatorSignature.expiry
-        });
+    ) external override onlyWhenNotPaused(PAUSER_OPERATOR_REGISTER_DEREGISTER_TO_OPERATOR_SETS) {
+        // Assert operator's signature has not expired.
+        require(operatorSignature.expiry >= block.timestamp, SignatureExpired());
+        // Assert `operator` is actually an operator.
+        require(delegation.isOperator(operator), OperatorNotRegistered());
+        // Assert that the AVS is an operator set AVS.
+        require(isOperatorSetAVS[msg.sender], InvalidAVS());
```

Comment:
so new AVSs will need to call `becomeOperatorSetAVS` before they can register any operators?
I think this is OK but definitely a potential mistake for people to make / something to call out in docs

Reply:
Does removing these checks introduce any concerns?

Removing them would be cheaper, and I don't think there are any negative implications. @gpsanant Thoughts?